### PR TITLE
Added migration for service user and permissions for admin group.

### PR DESCRIPTION
**Pull request by tasawernawaz**
ECOM-3420

Added migration for credentials service user and permission for admin group.

Please review this.
@awais786 @ahsan-ul-haq @zubair-arbi

### CODE REVIEW

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME).delete()
+    except User.DoesNotExist:
+        return
+
+
+def add_permissions_for_admins_group(apps, schema_editor):
+    """ Add permissions of add and update credentials for user_group admins. """
+    admins_group = Group.objects.get(name=Role.ADMINS)
+    permissions = Permission.objects.filter(codename__in=["add_usercredential", "change_usercredential"])
+    admins_group.permissions = permissions
```
Comment (ahsan-ul-haq) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: try doing 
`admins_group.permissions.add(permissions )`

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def clear_permissions_for_admins_group(apps, schema_editor):
+    """ Clears permissions of add and update credentials for user_group admins. """
+    admins_group = Group.objects.get(name=Role.ADMINS)
+    admins_group.permissions.clear()
```
Comment (ahsan-ul-haq) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: Add exception handling here we don't want migrations to fail.

File: credentials/settings/base.py [hunk 3]
```diff
@@ -252,7 +252,7 @@
 ORGANIZATIONS_CACHE_TTL = 0
 
 # Credentials service user in Programs service and LMS
-CREDENTIALS_SERVICE_USER = 'credentials_service_user'
+CREDENTIALS_SERVICE_USERNAME = 'credentials_service_user'
```
Comment (zubair-arbi) [inline credentials/settings/base.py]: remove it

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py (hunk 1 above)
Comment (tasawernawaz) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: We can't add a `queryset`.

**Comment by zubair-arbi** [comment]
lgtm :+1:

**Comment by tasawernawaz** [comment]
@jimabramson Please review this.

**Comment by tasawernawaz** [comment]
@clintonb Please review this.

**Comment by jimabramson** [comment]
@peter-fogg  could you take over my review on this one?

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        User.objects.get(username=settings.CREDENTIALS_SERVICE_USER).delete()
+    except User.DoesNotExist:
+        return
+
+
+def add_permissions_for_admins_group(apps, schema_editor):
+    """ Add permissions of add and update credentials for user_group admins. """
+    try:
+        admins_group = Group.objects.get(name=Role.ADMINS)
+    except Group.DoestNotExist:
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: Shouldn't the group be created if it doesn't exist?

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+from django.db import migrations
+from django.conf import settings
+from credentials.apps.core.constants import Role
+
+from credentials.apps.core.models import User
+
+
+def add_service_user(apps, schema_editor):
+    """Add service user."""
+    user, created = User.objects.get_or_create(username=settings.CREDENTIALS_SERVICE_USER, is_staff=True)
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: This user needs to be added to the admin goup.

**Comment by tasawernawaz** [comment]
@clintonb Please review it again.

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    if created:
+        user.set_unusable_password()
+        admin_group = Group.objects.get(name=Role.ADMINS)
+        user.groups.add(admin_group)
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: This should happen regardless of whether the user was created.

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        user.save()
+
+
+def remove_service_user(apps, schema_editor):
+    """Remove service user."""
+    try:
+        User.objects.get(username=settings.CREDENTIALS_SERVICE_USER).delete()
+    except User.DoesNotExist:
+        return
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: Use pass.

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def add_permissions_for_admins_group(apps, schema_editor):
+    """ Add permissions of add and update credentials for user_group admins. """
+    group, created = Group.objects.get_or_create(name=Role.ADMINS)
+    permissions = Permission.objects.filter(codename__in=["add_usercredential", "change_usercredential"])
+    group.permissions = permissions
+
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: group.save()

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+def clear_permissions_for_admins_group(apps, schema_editor):
+    """ Clears permissions of add and update credentials for user_group admins. """
+    try:
+        admins_group = Group.objects.get(name=Role.ADMINS)
+    except Group.DoesNotExist:
+        return
+
+    admins_group.permissions.clear()
+
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: group.save()

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0002_auto_20160111_1251'),
+    ]
+
+    operations = [
+        migrations.RunPython(code=add_service_user, reverse_code=remove_service_user),
+        migrations.RunPython(code=add_permissions_for_admins_group, reverse_code=clear_permissions_for_admins_group),
```
Comment (clintonb) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: Create the group/permissions first.

**Comment by clintonb** [comment]
If you haven't already, make sure you run this locally and verify the results.

**Comment by tasawernawaz** [comment]
@clintonb Yes I have verified it locally, can you please tale another look.

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py (hunk 5 above)
Comment (peter-fogg) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: The `created` variable is unused; replace it with `__`.

File: credentials/apps/core/migrations/0003_auto_20160114_1001.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    """Remove service user."""
+    try:
+        User.objects.get(username=settings.CREDENTIALS_SERVICE_USER).delete()
+    except User.DoesNotExist:
+        pass
+
+
+def add_permissions_for_admins_group(apps, schema_editor):
+    """ Add permissions of add and update credentials for user_group admins. """
+    group, created = Group.objects.get_or_create(name=Role.ADMINS)
```
Comment (peter-fogg) [inline credentials/apps/core/migrations/0003_auto_20160114_1001.py]: Similar note about `created` here.

**Comment by peter-fogg** [comment]
:+1: Once the tests pass.

**Pull request merged**